cmd/create/cluster/cmd.go

```diff
@@ -3444,23 +3444,32 @@ func getExpectedResourceIDForAccRole(hostedCPPolicies bool, roleARN string, role
 	return strings.ToLower(fmt.Sprintf("%s-%s-Role", rolePrefix, accountRoles[roleType].Name)), rolePrefix, nil
 }
 
-func getInitialValidSubnets(awsClient aws.Client, reporter *reporter.Object) ([]*ec2.Subnet, error) {
+func getInitialValidSubnets(aws aws.Client, ids []string, r *reporter.Object) ([]*ec2.Subnet, error) {
 	initialValidSubnets := []*ec2.Subnet{}
 	excludedSubnets := []string{}
-	allSubnets, err := awsClient.GetSubnetIDs()
+
+	var validSubnets []*ec2.Subnet
+	var err error
+
+	if len(ids) == 0 {
```

Comment:
If this is checked internally there's no need to have this separation when calling